File: Kotlin/src/main/java/no/agitec/fagaften/mars/kotlin/section02/convert/textadventuregamekotlin/java/Locations.java
package no.agitec.fagaften.mars.kotlin.section02.convert.textadventuregamekotlin.java;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Locations {

    public static Map<Integer, Location> locations = new HashMap();

    static {
        Scanner scanner = null;

        // Read locations
        try {
            scanner = new Scanner(new FileReader(getFilePath() + "locations_big.txt"));
            scanner.useDelimiter(",");
            while(scanner.hasNextLine()) {
                int loc = scanner.nextInt();
                scanner.skip(scanner.delimiter());
                String description = scanner.nextLine();
                locations.put(loc, new Location(loc, description));
            }

        }
        catch(IOException e) {
            e.printStackTrace();
        }
        finally {
            if(scanner != null) {
                scanner.close();
            }
        }

        // Now read the exits
        try {
            scanner = new Scanner(new BufferedReader(new FileReader(getFilePath() + "directions_big.txt")));
            scanner.useDelimiter(",");
            while(scanner.hasNextLine()) {
                String input = scanner.nextLine();
                String[] data = input.split(",");
                int loc = Integer.parseInt(data[0]);
                String direction = data[1];
                int destination = Integer.parseInt(data[2]);

                Location location = locations.get(loc);
                location.addExit(direction, destination);
            }
        }
        catch (IOException e) {
            e.printStackTrace();
        }
        finally {
            if(scanner != null) {
                scanner.close();
            }
        }
    }

    private static String getFilePath() {
        String filePath = new File("").getAbsolutePath();
        return filePath + "\\src\\main\\java\\no\\agitec\\fagaften\\mars\\kotlin\\section2\\textadventuregamejava\\files-java\\";
    }
}

File: Kotlin/src/main/java/no/agitec/fagaften/mars/kotlin/section02/convert/textadventuregamekotlin/kotlin/Locations.kt
package no.agitec.fagaften.mars.kotlin.section02.convert.textadventuregamekotlin.kotlin

import java.io.File


fun readLocationInfo() : Map<Int, Location> {

    val locations = mutableMapOf<Int, Location>()

    File(getFilePath() + "locations_big.txt").reader().forEachLine {
        val tokens = it.split("`")

        val location = Location(tokens[0].toInt(), tokens[1])
        locations[location.locationID] = location
    }

    File(getFilePath() + "directions_big.txt").reader().forEachLine {
        val tokens = it.split(",")

        locations[tokens[0].toInt()]?.addExit(tokens[1], tokens[2].toInt())
    }

    return locations

}

private fun getFilePath(): String {
    val filePath = File("").absolutePath
    var path = "$filePath\\src\\main\\java\\no\\agitec\\fagaften\\mars\\kotlin\\section2\\convert\\textadventuregamekotlin\\files-kotlin\\"
    //System.out.println("** " + path + " **");
    return path
}